### DISCUSSION TITLE
Many customers with same source

### DISCUSSION THREAD

**Discussion by cristian-araujo** [devaslanphp/help-desk#82] [Ideas]
One of the possibilities that a Help Desk should have is to be able to provide support to different companies on the same platform.

Each company should have projects in which their employees can have access to the support platform.

The relationship would be the following, I made it fast so something can be wrong

![image](https://user-images.githubusercontent.com/72935660/192118632-e8e651ca-a36c-4449-a76f-b4098d8f469e.png)

In this way, help desk can has many companies with projects and employees

**Comment by heloufir** [comment]
That's a good idea, I will make it a task in the Help Desk project.

Thank you it's clear.

**Comment by heloufir** [comment]
Hello @cristian-araujo 

Just to let you know a new Release (v1.4.0) has been published containing the companies management system and `spatie/laravel-permission` integration

It's a little bit a big change on the application so if you find any issue you can create an issue so I can resolve it.

**Comment by cristian-araujo** [comment]
great, i'm checking it, it's good but I think something are missing.


how can I set a project to a company? 
    - the administrator should be able to generate a project to a company
    - inside the Porject table the owner_id can be changed to company_id or you can save owner_id if you want a log of who's created it
how can I add a company user to a project?
    - pivot table to manage company users with projects.
    
a Company can has many projects. 
   - it would be Company -> hasMany -> Projects
   - Project -> belongsTo -> Company
   
a Project can have many company users, the company can select who employees can have access to its project.

when an employee login the listed projects need to be only users that have access.

I never worked before with livewire but I'll try to help you with the improvements.

could you re-open the discuss of roles and permissions? I can't see it, i've some things about it

**Reply by heloufir** [reply]
For now, the project can be created by a user (this user of course can be the responsible of the company and have the permission to create projects).
But for the next releases I am planning to add the link between Project and Company of course.

And for the employees having access to projects or not, you got the point, I will make sure to add this too on the next releases.

But for now, the functions added on this Release (v1.4.0) are:
- Manage companies
- Link a responsible to the company (it can be the Customer administrator, for example)
- This company responsible can add employees and link them to one of the companies he owns
- The users have permissions so a User having the permission to assign permissions to users can attach them 

But I think to make it easy to manage, I will include the roles management to the next releases too, so the users can be attached to roles instead of directly attach them to roles.

**Reply by heloufir** [reply]
For the discussion about the roles and permissions, I think you can create an issue this will be more easy to manage it there and link branches / PR.

**Reply by heloufir** [reply]
And for this discussion too I think it's better too to create an issue for improvements

**Reply by heloufir** [reply]
You can use this two issues:
- For managing companies #95 
- For roles and permissions #96
